### PR TITLE
Valera Fedorenko ethse tests

### PR DESCRIPTION
**Pull request by valeraf23**
Task 3

### CODE REVIEW

**Comment by valeraf23** [comment]
столько комитов из-за того , что у меня был трабл с npm run test.
До того как получил совет переименовать truffle в truffle-config --  я проверял выполнение тестов через travis-ci.

File: contracts/lendOrder.sol [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    function repayMoney(uint _amount, address _debtorAddress) public onlyOwner() moreThanZero(_amount) {
+        require(_debtorAddress > 0x0);
+        require(balanceByAddr[_debtorAddress] >= _amount);
+        uint id = orders.push(Order(userNameByAddr[_debtorAddress],_amount, uint32(now),"repay")) - 1;
+        balanceByAddr[_debtorAddress] = _safeSub( balanceByAddr[_debtorAddress], _amount);
+        NewOrder(id, _amount);
+    }
+
+    function showBalance() public onlyKnownUser() constant returns (uint) {
```
Comment (Purka) [inline contracts/lendOrder.sol:35]: what will happen if I'm not registered and try to look someones balance?

File: test/lendOrder.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  it('should allow to regist new user', async () => {
+    const borrower = accounts[4];
+    const newUser = 'testUser';
+    let result = await lendOrder.register(newUser,{from: borrower});
+    assert.equal(result.logs.length, 1);
+    assert.equal(result.logs[0].event, 'NewUser');
+    assert.equal(result.logs[0].args._userName, newUser);
+  });
+
+  it('should emit Repayed event on repay', async () => {
```
Comment (Purka) [inline test/lendOrder.js]: The name of the test doesn't match with content

File: test/lendOrder.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  });
+
+  it('should fail on overflow when borrowing', async () => {
+    const borrower = accounts[3];
+    const value = '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+    await lendOrder.lendMoney(value, {from: borrower});
+    await asserts.throws(lendOrder.lendMoney(1, {from: borrower}));
+  });
+
+  it('should emit Borrowed event on borrow',async () => {
```
Comment (Purka) [inline test/lendOrder.js]: The name of the test doesn't match with content

File: test/lendOrder.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  it('should not allow owner to borrow',async () => {
+    const value = 1000;
+        await asserts.throws(lendOrder.lendMoney(value, {from: OWNER}));
+  });
+
+  it('should not allow not owner to repay', async () => {
+    const borrower = accounts[3];
+    const value = 1000;
+    await asserts.throws(lendOrder.repayMoney(value, borrower, {from: borrower}));
+  });
```
Comment (Purka) [inline test/lendOrder.js:96]: Do you check that borrower can't borrow zero and owner can't repay zero? 
That registered borrower can borrow and not registered can't?
Some more test?

Comment (valeraf23) [reply test/lendOrder.js:96]: I added moreThanZero modifier

but ok  -- I am going add several tests

File: test/lendOrder.js (hunk 3 above)
Comment (valeraf23) [reply test/lendOrder.js]: change to == > should emit NewOrder event on borrow

File: test/lendOrder.js (hunk 2 above)
Comment (valeraf23) [reply test/lendOrder.js]: change to == > should emit NewOrder event on repay

File: contracts/lendOrder.sol (hunk 1 above)
Comment (valeraf23) [reply contracts/lendOrder.sol:35]: I see
I have added private to balanceByAddr

**Comment by valeraf23** [comment]
Look please again

**Comment by valeraf23** [comment]
Could you look again, please?

**Review by Purka — APPROVED** [review]

**Pull request closed without merging**